### PR TITLE
Add ePBS to db

### PR DESCRIPTION
**Pull request by terencechain**
This PR updates DB pkg to be ePBS compatible. Mainly these three things
1.) Block
2.) Beacon state
   - Marshal method was missing. I added the the correct name to build file
3.) Exuection Payload Envelope (blind version)
   - New protobuf `SignedExecutionPayloadEnvelopeBlind` was defined. The payload itself is replaced by `payload_root`
   - Getter is `SignedExecutionPayloadEnvelopeBlind(ctx context.Context, blockRoot []byte)` which uses beacon block root to retrieve
   - Setter is `SaveSignedExecutionPayloadEnvelopeBlind(ctx context.Context, envelope *ethpb.SignedExecutionPayloadEnvelopeBlind)`

### CODE REVIEW

File: beacon-chain/db/kv/state.go [hunk 1]
```diff
@@ -222,114 +222,146 @@ func (s *Store) saveStatesEfficientInternal(ctx context.Context, tx *bolt.Tx, bl
...
-			if err != nil {
-				return err
-			}
-			encodedState := snappy.Encode(nil, append(altairKey, rawObj...))
-			if err := bucket.Put(rt[:], encodedState); err != nil {
-				return err
-			}
-			pbState.Validators = valEntries
-			if err := valIdxBkt.Put(rt[:], validatorKeys[i]); err != nil {
+			if err := s.processAltair(ctx, rawType, rt[:], bucket, valIdxBkt, validatorKeys[i]); err != nil {
```
Comment (terencechain) [inline beacon-chain/db/kv/state.go:231]: All these is to fix the cognitive complexity for golang ci lint complain

File: beacon-chain/db/iface/interface.go [hunk 2]
```diff
@@ -87,6 +88,7 @@ type NoHeadAccessDatabase interface {
 	SaveDepositContractAddress(ctx context.Context, addr common.Address) error
 	// SaveExecutionChainData operations.
 	SaveExecutionChainData(ctx context.Context, data *ethpb.ETH1ChainData) error
+	SaveSignedExecutionPayloadEnvelopeBlind(ctx context.Context, envelope *ethpb.SignedExecutionPayloadEnvelopeBlind) error
```
Comment (potuz) [inline beacon-chain/db/iface/interface.go:91]: Do we really need the Blind here? and even such a long name? any problem in calling this `SavePayloadEnvelope`?

File: beacon-chain/db/kv/blocks.go [hunk 3]
```diff
@@ -303,7 +303,7 @@ func prepareBlockBatch(blks []blocks.ROBlock, shouldBlind bool) ([]blockBatchEnt
 				if !errors.Is(err, blocks.ErrUnsupportedVersion) {
 					return nil, errors.Wrapf(err, "could not convert block to blinded format for root %#x", batch[i].root)
 				}
-				// Pre-deneb blocks give ErrUnsupportedVersion; use the full block already in the batch entry.
+				// Pre-bellatrix and ePBS blocks give ErrUnsupportedVersion; use the full block already in the batch entry.
```
Comment (potuz) [inline beacon-chain/db/kv/blocks.go:306]: This comment change is confusing

File: beacon-chain/db/kv/blocks_test.go [hunk 4]
```diff
@@ -857,7 +869,7 @@ func TestStore_BlocksBySlot_BlockRootsBySlot(t *testing.T) {
 			require.NoError(t, err)
 			assert.Equal(t, true, proto.Equal(wantedPb, retrieved0Pb), "Wanted: %v, received: %v", retrievedBlocks[0], wanted)
 			wanted = b3
-			if b3.Version() >= version.Bellatrix {
+			if b3.Version() >= version.Bellatrix && b3.Version() < version.EPBS {
```
Comment (potuz) [inline beacon-chain/db/kv/blocks_test.go:872]: Don't we need to add then a test for epbs?

File: proto/eth/v1/generated.ssz.go [hunk 5]
```diff
@@ -1,5 +1,5 @@
 // Code generated by fastssz. DO NOT EDIT.
-// Hash: d06a72227c2f5e350916cce3e89f4e855135a2a22f6ea263dedc68fa506c1ba7
```
Comment (potuz) [inline proto/eth/v1/generated.ssz.go:2]: This file is better not to add the change

File: proto/eth/v2/generated.ssz.go [hunk 6]
```diff
@@ -1,5 +1,5 @@
 // Code generated by fastssz. DO NOT EDIT.
-// Hash: 6f33097dd41d3dd49d35b7cfceef5a1afca20f05d65b18215748b459db16f99b
+// Hash: 6b214399116c0ca31026da23db2b85fccd78e16d7b9113c83b4a9ee4e60977f6
```
Comment (potuz) [inline proto/eth/v2/generated.ssz.go:2]: This file is better not to add the change

File: proto/prysm/v1alpha1/blind_payload_envelope.proto [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+option java_outer_classname = "BlindPayloadEnvelopeProto";
+option java_package = "org.ethereum.eth.v1alpha1";
+option php_namespace = "Ethereum\\Eth\\v1alpha1";
+
+message SignedExecutionPayloadEnvelopeBlind {
+    ExecutionPayloadEnvelopeBlind message = 1;
+    bytes signature 		 = 2 [(ethereum.eth.ext.ssz_size) = "96"];
+}
+
+message ExecutionPayloadEnvelopeBlind {
```
Comment (potuz) [inline proto/prysm/v1alpha1/blind_payload_envelope.proto:32]: I think we probably can start naming these PayloadEnvelopeBlind or BlindPayloadEnvelope

File: proto/prysm/v1alpha1/blind_payload_envelope.proto [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+import "proto/eth/ext/options.proto";
+
+option csharp_namespace = "Ethereum.Eth.v1alpha1";
+option go_package = "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1;eth";
+option java_multiple_files = true;
+option java_outer_classname = "BlindPayloadEnvelopeProto";
+option java_package = "org.ethereum.eth.v1alpha1";
+option php_namespace = "Ethereum\\Eth\\v1alpha1";
+
+message SignedExecutionPayloadEnvelopeBlind {
```
Comment (potuz) [inline proto/prysm/v1alpha1/blind_payload_envelope.proto:27]: Same with this, probably can be simply named `SignedPayloadEnvelopeBlind`  or `SignedBlindPayloadEnvelope`

File: testing/util/random/epbs.go [hunk 9]
```diff
@@ -119,8 +120,12 @@ func AttestationData(t *testing.T) *ethpb.AttestationData {
 
 // Deposit creates a random Deposit for testing purposes.
 func Deposit(t *testing.T) *ethpb.Deposit {
+	proof := make([][]byte, 33)
+	for i := 0; i < 33; i++ {
+		proof[i] = randomBytes(32, t)
+	}
 	return &ethpb.Deposit{
-		Proof: [][]byte{randomBytes(32, t), randomBytes(32, t), randomBytes(32, t)},
```
Comment (potuz) [inline testing/util/random/epbs.go:123]: why before it was 33 and now 32 the length of the slice?

File: beacon-chain/db/kv/blocks.go (hunk 3 above)
Comment (terencechain) [reply beacon-chain/db/kv/blocks.go:306]: I'll revert this change. There's nothing we have to do here anyway

File: beacon-chain/db/kv/blocks_test.go (hunk 4 above)
Comment (terencechain) [reply beacon-chain/db/kv/blocks_test.go:872]: I'm unclear what you meant. We have an EPBS test in `blockTests`. These changes fix the previous behavior where the test always assumes the saved block is blinded for b3.

File: testing/util/random/epbs.go (hunk 9 above)
Comment (terencechain) [reply testing/util/random/epbs.go:123]: Before we didn't use the correct length. Now we are using correct length 33 or else HTR fails

**Pull request merged**